### ISSUE TITLE
Version 1.2 decoding regression

### ISSUE THREAD

**Issue by jangernert** [feed-rs/feed-rs#172]
I noticed that decoding of RSS 1.0 HTML content doesn't work anymore with version 1.2. I used the `golem.de` [feed](http://rss.golem.de/rss.php?feed=RSS1.0) to test this with the `testurls` program. It uses `ISO-8859-1` encoding.

```
<?xml version="1.0" encoding="ISO-8859-1"?>
```

Before:

```
"<img src=\"https://www.golem.de/2211/170034-355606-355601_rc.jpg\" width=\"140\" height=\"140\" vspace=\"3\" hspace=\"8\" align=\"left\">Müssten die nicht gleich mit dabei sein? The Callisto Protocol bekommt nach der Veröffentlichung kostenpflichtige Kampagneninhalte. (<a href=\"https://www.golem.de/specials/the-callisto-protocol/\">The Callisto Protocol</a>, <a href=\"https://www.golem.de/specials/survival/\">Survival</a>) <img src=\"https://cpx.golem.de/cpx.php?class=17&amp;aid=170034&amp;page=1&amp;ts=1669369502\" alt=\"\" width=\"1\" height=\"1\" />"
```

Now:

```
"<img src=\"https://www.golem.de/2211/170034-355606-355601_rc.jpg\" width=\"140\" height=\"140\" vspace=\"3\" hspace=\"8\" align=\"left\">MÃ¼ssten die nicht gleich mit dabei sein? The Callisto Protocol bekommt nach der VerÃ¶ffentlichung kostenpflichtige Kampagneninhalte. (<a href=\"https://www.golem.de/specials/the-callisto-protocol/\">The Callisto Protocol</a>, <a href=\"https://www.golem.de/specials/survival/\">Survival</a>) <img src=\"https://cpx.golem.de/cpx.php?class=17&amp;aid=170034&amp;page=1&amp;ts=1669369502\" alt=\"\" width=\"1\" height=\"1\" />"
```

TL;DR:  `Müssten` with correct german umlaut before now is `MÃ¼ssten`

I tracked the regression down to bf2a5cdc7cf461ce2bb0c648233f8248308c506e.

edit:

the description still gets decoded properly `<description>Müssten ...`.
the problem is `<content:encoded><![CDATA[<img src="https://www.golem.de/2211/170034-355606-355601_rc.jpg" width="140" height="140" vspace="3" hspace="8" align="left">Müssten ...`.

edit2:

~~stepping through `quick-xml` I noticed the encoding is `windows-1252` instead of `ISO-8859-1` although they seem to be nearly identical. So not sure if that is an issue.~~
Encoding was `windows-1252` with the old version as well.

edit3:

decoding in `quick-xml` is now done with
```
encoding
        .decode_without_bom_handling_and_without_replacement(bytes)
        .ok_or(Error::NonDecodable(None))
```
instead of `self.encoding.decode(bytes).0` in the old version. Not sure if that is the culprit.

edit4:

This might be a case of double decoding text again. I saved the raw bytes that get fed into [this line](https://github.com/feed-rs/feed-rs/blob/master/feed-rs/src/xml/mod.rs#L543) to a file and opened them up with a text editor. Loading the file as Utf8 renders all the letters correctly. Forcing to load the file with `ISO-8859-1` has corrupt umlauts again.

Then I replaced `let escaped_text = reader.decoder().decode(text)?;` with a simple
```
let escaped_text = std::str::from_utf8(text).map_err(|e| XmlError::Parser { e: quick_xml::Error::NonDecodable(Some(e)) })?;
```
and now `testurls` also displays everything fine.

So my guess is that `quick-xml` does some decoding already and feed-rs tries to decode it again [here](https://github.com/feed-rs/feed-rs/blob/master/feed-rs/src/xml/mod.rs#L543).
